posthog/api/capture.py

```diff
@@ -379,20 +379,28 @@ def drop_events_over_quota(token: str, events: list[Any]) -> EventsOverQuotaResu
                 if settings.QUOTA_LIMITING_ENABLED:
                     recordings_were_limited = True
                     continue
-
         else:
+            # Regular events - check if they're free events (i.e. events that we don't quota-limit) first
+            event_name = event.get("event")
```

Comment:
Problems
- [ ] This is only Django capture - rust capture is missing
- [ ] This should be environment variable based and I think we should keep it hyper specific to the list of events, not prefixes as that opens it up to huge abuse potential. 
- [ ] The plugin server also drops data if the quota is hit so this needs to be abstracted to env vars and ideally centralised